mobile_wallet/src/lib.rs

```diff
@@ -679,6 +779,127 @@ fn create_credential_aux(input: &str) -> anyhow::Result<String> {
+        Left(_) => AccountAddress::new(&cdi.values.cred_id),
+        Right(address) => address,
+    };
+
+    // unwrap is safe here since we've generated the credential already, and that
+    // does the same computation.
+    let enc_key = id_use_data.aci.prf_key.prf_exponent(acc_num).unwrap();
+    let secret_key = elgamal::SecretKey {
+        generator: *global_context.elgamal_generator(),
+        scalar:    enc_key,
+    };
+
+    let credential_message = AccountCredentialMessage {
+        message_expiry: expiry,
+        credential:     AccountCredential::Normal { cdi },
+    };
+
+    let response = json!({
+        "credential": Versioned::new(VERSION_0, credential_message),
+        "commitmentsRandomness": randomness,
+        "accountKeys": AccountKeys::from(cred_data),
+        "encryptionSecretKey": secret_key,
+        "encryptionPublicKey": elgamal::PublicKey::from(&secret_key),
+        "accountAddress": address,
+    });
+    Ok(to_string(&response)?)
+}
+
+/// For deterministic credential creation using an identity object containing a
+/// version 1 pre-identity object, i.e., no initial account involved.
+fn create_credential_v1_aux(input: &str) -> anyhow::Result<String> {
+    let v: Value = from_str(input)?;
+    let expiry = try_get(&v, "expiry")?;
+    let ip_info: IpInfo<Bls12> = try_get(&v, "ipInfo")?;
+
+    let ars_infos: BTreeMap<ArIdentity, ArInfo<ExampleCurve>> = try_get(&v, "arsInfos")?;
+
+    let global_context: GlobalContext<ExampleCurve> = try_get(&v, "global")?;
+
+    let id_object: IdentityObjectV1<Bls12, ExampleCurve, AttributeKind> =
+        try_get(&v, "identityObject")?;
+
+    let tags: Vec<AttributeTag> = try_get(&v, "revealedAttributes")?;
+
+    let wallet = parse_wallet_input(&v)?;
+    let identity_index: u8 = try_get(&v, "identityIndex")?;
+    let acc_num: u8 = try_get(&v, "accountNumber")?;
+
+    let sig_retrievel_randomness: ps_sig::SigRetrievalRandomness<Bls12> =
+        wallet.get_blinding_randomness(identity_index as u32)?;
+    let id_cred_sec: PedersenValue<ExampleCurve> =
+        PedersenValue::new(wallet.get_id_cred_sec(identity_index as u32)?);
+    let id_cred: IdCredentials<ExampleCurve> = IdCredentials { id_cred_sec };
+    let chi = CredentialHolderInfo::<ExampleCurve> { id_cred };
+    let prf_key: prf::SecretKey<ExampleCurve> = wallet.get_prf_key(identity_index as u32)?;
+    let aci = AccCredentialInfo {
+        cred_holder_info: chi,
+        prf_key,
+    };
+    let id_use_data = IdObjectUseData {
+        aci,
+        randomness: sig_retrievel_randomness,
+    };
+
+    // The mobile wallet for now only creates new accounts and does not support
+    // adding credentials onto existing ones. Once that is supported the address
+    // should be coming from the input data.
+    let new_or_existing = Left(expiry);
+
+    // The mobile wallet can only create new accounts, which means new credential
+    // data will be generated.
```

Comment:
This comment kind of duplicates what the one just above says. Perhaps we can combine them into one.